### ISSUE TITLE
Add Opening Proposal Limits Discussion

### ISSUE THREAD

**Issue by bwhm** [Joystream/joystream#780]
After a talk with @shamil-gadelshin, we encountered some unresolved questions with this proposal. It takes quite a lot of inputs, and is probably not the most user friendly.

### `activate_at: hiring::ActivateOpeningAt`
If the proposal is voted through, this decides when the `AcceptingApplications` stage begins.

`CurrentBlock` means the opening starts in the `AcceptingApplications` stage from the block the proposal is executed.

`ExactBlock` takes a specific block height, as input, which makes sense when a lead or sudo creates the opening. When it goes via the proposal system, it doesn't really work as well. Suppose the `voting_period` of said proposal is 7 days, and the `grace_period` is 2 days. The proposer sets a block height 9 days in to the future. If the proposal is voted through immediately, it will be 7 days in the `WaitingToBegin` stage, whereas if it's voted through at the end, it will be 0 days.

Suppose under the same conditions, the proposer sets a block height 5 days in to the future. If the voting lasts > 3 days, I assume the execution would fail, or in best case, start in the `AcceptingApplications` stage.

TL;DR
If one sets an exact block, you have little impact on how long the opening will be created and stay in the `WaitingToBegin` stage, and might see the execution fail.

Assuming we keep the extrinsic as is, we need to check that for `ExactBlock`, `min > voting_period+grace_period`.

There are two alternatives to this:
1. Don't allow `ExactBlock`, and enforce `CurrentBlock` if it's a proposal. For sudo and lead, it can stay as is. If we choose this option, there is no point in making the "Accept Applications" proposal type.
2. Instead of `ExactBlock`, replace with the input `MaxWaitingToBeginPeriodLength`, that takes a block number, and means the number of blocks after execution of the proposal. It would work similar to the `max_review_period_length`.
@bedeho?

### `commitment: OpeningPolicyCommitment`
This takes ton of options, which does not all make sense.

#### `application_rationing_policy`
Although this input has some value, I'm not sure if we need any runtime checks here except `min>0`. It may make sense to be more opinionated on the GUI side.

#### `max_review_period_length`
After thinking about it, I think it's better to keep the runtime simple, and only ensure `min>0` in blocks. 

For pioneer, I would say something like:
```
min>43_200` blocks -> three days
max>201_600` blocks -> two weeks
```

#### `application_staking_policy` & `role_staking_policy`
No runtime checks here except:
```
amount (all four iterations):
min>0

*_period_length (all four iterations, if included)
min>0
```

#### `role_slashing_terms`
As I understood @shamil-gadelshin, this is not actually read anywhere, and defaults to `Slashable`. 

For a lead, it makes no sense to have them not being slashable anyway, but there may be value in allowing the user to include `max_count` and `max_percent_pts_per_time`. If it's confirmed that these are not read, I suppose we skip checks for Nicaea.

#### `*_application_stake_unstaking_period`
There are four of these:
- `fill_opening_successful_applicant_application_stake_unstaking_period`
- `fill_opening_failed_applicant_application_stake_unstaking_period`
- `terminate_lead?_application_stake_unstaking_period`
- `exit_lead?_role_application_stake_unstaking_period`

I think we should enforce leaving them empty (e.g. 0) in pioneer for Nicaea, and don't perform any checks.

#### `*_role_stake_unstaking_period`
There are three of these:
- `fill_opening_failed_applicant_role_stake_unstaking_period`
I think we should enforce leaving this empty (e.g. 0) in pioneer for Nicaea, and don't perform any checks.

- `terminate_lead?_role_stake_unstaking_period`
- `exit_lead?_role_stake_unstaking_period`
These can have some value including, and I think it's worth allowing the user to set them in pioneer. The runtime should at the very least ensure `min>0`. I think it's enough if the max is set in pioneer. 
Something like `max>201_600` blocks -> two weeks.

### `human_readable_text: Bytes`
This should have the same limits as for the "Text Proposal" text. For the I don't think we should reuse the json schema for the curator role.

---

Not adding this to the project yet, as this issue is for discussion purposes.
